mex/common/wikidata/extract.py

```diff
@@ -11,31 +11,41 @@
 
 def search_organization_by_label(
     item_label: str,
+    lang: TextLanguage = TextLanguage.EN,
 ) -> WikidataOrganization | None:
-    """Search for an item in wikidata. Only organizations are fetched.
+    """Search for an item in wikidata. Only organizations are searched.
 
     Args:
         item_label: Item title or label to be searched
+        lang: lang in which item should be searched. Default: TextLanguage.EN
 
     Returns:
-        WikidataOrganization if only one organization is found
-        None if no or multiple organizations are found
+        WikidataOrganization if organization is found
+        None if no organization is found
     """
     connector = WikidataQueryServiceConnector.get()
     item_label = item_label.replace('"', "")
     query_string = (
         "SELECT distinct ?item ?itemLabel ?itemDescription "
-        "WHERE{"
-        "?item (wdt:P31/wdt:P8225*/wdt:P279*) wd:Q43229."
-        f'?item ?label "{item_label}"@en.'
-        "?article schema:about ?item ."
-        'SERVICE wikibase:label { bd:serviceParam wikibase:language "en". }'
-        "}"
+        "WHERE { "
+        "SERVICE wikibase:mwapi { "
+        'bd:serviceParam wikibase:api "EntitySearch" . '
+        'bd:serviceParam wikibase:endpoint "www.wikidata.org" . '
+        f'bd:serviceParam mwapi:search "{item_label}" . '
+        f'bd:serviceParam mwapi:language "{lang}" . '
+        "?item wikibase:apiOutputItem mwapi:item . "
+        "?num wikibase:apiOrdinal true . "
+        "} "
+        "?item (wdt:P31/wdt:P8225*/wdt:P279*) wd:Q43229. "
+        'SERVICE wikibase:label { bd:serviceParam wikibase:language "[AUTO_LANGUAGE],en,de". } '  # noqa: E501
+        "} "
+        "ORDER BY ASC(?num) "
+        "LIMIT 1 "
```

Comment:
Could you explain what the WHERE clause is doing? I don't understand it

Reply:
its part of the MediaWiki API Query Service. Its faster and has better caching than the older query we were using here. I am not sure how much technically superior it is.

WHERE clause is combining some parameters and requesting internal MediaWiki API, specifying:
- Which action to use: `EntitySearch`
- Which endpoint to use: `wikidata.org` they have wikipedia as well. 
- What to search: `item_label`
- Which language to search in: `lang`
- What do we expect it to return: `item` an organization or person is an item in wikidata
- apiOrdinal is used for ordering the results to get the first best match

you can read more about it here: https://www.mediawiki.org/wiki/Wikidata_Query_Service/User_Manual/MWAPI